tests/ops/qutrit/test_qutrit_parametric_ops.py

```diff
@@ -0,0 +1,477 @@
+        assert np.allclose(
+            op(theta, wires=0, subspace=subspace).matrix(), expected, atol=tol, rtol=0
+        )
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("op, theta, subspace, expected", matrix_data)
+    def test_matrix_tf(self, op, theta, subspace, expected, tol):
+        """Test that compute_matrix works with tensorflow variables"""
+        import tensorflow as tf
+
+        theta = tf.Variable(theta)
+        expected = tf.convert_to_tensor(expected)
+        assert qml.math.allclose(
+            op.compute_matrix(theta, subspace=subspace), expected, atol=tol, rtol=0
+        )
+        assert qml.math.allclose(
+            op(theta, wires=0, subspace=subspace).matrix(), expected, atol=tol, rtol=0
+        )
+
+
+label_data = [
+    (qml.TRX(1.23456, wires=0), "TRX", "TRX\n(1.23)", "TRX\n(1)", "TRX\n(1)†"),
+]
+
+label_data_broadcasted = [
+    (qml.TRX(np.array([1.23, 4.56]), wires=0), "TRX", "TRX", "TRX", "TRX†"),
+]
+
+
+class TestLabel:
+    """Test the label method on parametric ops"""
+
+    @pytest.mark.parametrize("op, label1, label2, label3, label4", label_data)
+    def test_label_method(self, op, label1, label2, label3, label4):
+        """Test label method with plain scalars."""
+
+        assert op.label() == label1
+        assert op.label(decimals=2) == label2
+        assert op.label(decimals=0) == label3
+
+        op = qml.adjoint(op)
+        assert op.label(decimals=0) == label4
+
+    @pytest.mark.parametrize("op, label1, label2, label3, label4", label_data_broadcasted)
+    def test_label_method_broadcasted(self, op, label1, label2, label3, label4):
+        """Test broadcasted label method with plain scalars."""
+
+        assert op.label() == label1
+        assert op.label(decimals=2) == label2
+        assert op.label(decimals=0) == label3
+
+        op = qml.adjoint(op)
+        assert op.label(decimals=0) == label4
+
+    @pytest.mark.tf
+    def test_label_tf(self):
+        """Test label methods work with tensorflow variables"""
+        import tensorflow as tf
+
+        op1 = qml.TRX(tf.Variable(0.123456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(0.12)"
+
+    @pytest.mark.torch
+    def test_label_torch(self):
+        """Test label methods work with torch tensors"""
+        import torch
+
+        op1 = qml.TRX(torch.tensor(1.23456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(1.23)"
+
+    @pytest.mark.jax
+    def test_label_jax(self):
+        """Test the label method works with jax"""
+        import jax
+
+        op1 = qml.TRX(jax.numpy.array(1.23456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(1.23)"
+
+    def test_string_parameter(self):
+        """Test labelling works if variable is a string instead of a float."""
+
+        op1 = qml.TRX("x", wires=0)
+        assert op1.label() == "TRX"
+        assert op1.label(decimals=0) == "TRX\n(x)"
+
+    def test_string_parameter_broadcasted(self):
```

Comment:
do string parameters.. mean anything?

Reply:
No, but this test is pretty much a copy from the qubit parametric ops test file.

Reply:
seems kinda like we're just testing that python isn't a typed language which doesn't add much to PennyLane reliability.. I guess it can stay if it already exists in qubit op tests, but I'd probably remove it from there instead lol